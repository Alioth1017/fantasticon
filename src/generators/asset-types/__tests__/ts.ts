import tsGen from '../ts';

const mockAssets = { foo: {}, bar: {} };

const mockCodepoints = { foo: 'oof', bar: 'baz' };

const mockOptions = {
  name: 'my-icons-set',
  assets: mockAssets,
  codepoints: mockCodepoints
} as any;

const getCleanGen = async () =>
  ((await tsGen.generate(mockOptions, null)) as string)
    .replace(/\n+/, '')
    .replace(/\s+/g, ' ');

describe('`TS` asset generator', () => {
  test('renders expected TypeScript module content', async () => {
    expect(await tsGen.generate(mockOptions, null)).toMatchSnapshot();
  });

  test('correctly renders type declaration', async () => {
    expect(await getCleanGen()).toContain(
      'export type MyIconsSetId = | "foo" | "bar";'
    );
  });

  test('correctly enum declaration', async () => {
    expect(await getCleanGen()).toContain(
      'export enum MyIconsSet { Foo = "foo", Bar = "bar", }'
    );
  });

  test('correctly codepoints declaration', async () => {
    expect(await getCleanGen()).toContain(
      'export const MY_ICONS_SET_CODEPOINTS: { [key in MyIconsSet]: string }' +
        ' = { [MyIconsSet.Foo]: "oof", [MyIconsSet.Bar]: "baz", };'
    );
  });
});
