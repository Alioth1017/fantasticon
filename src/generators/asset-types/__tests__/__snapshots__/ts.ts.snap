// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`TS\` asset generator renders expected TypeScript module content 1`] = `
"export enum MyIconsSet {
  Foo = \\"foo\\",
  Bar = \\"bar\\",
}

export const MY_ICONS_SET_CODEPOINTS: { [key in MyIconsSet]: string } = {
  [MyIconsSet.Foo]: \\"oof\\",
  [MyIconsSet.Bar]: \\"baz\\",
};

export type MyIconsSetId =
  | \\"foo\\"
  | \\"bar\\";
"
`;
